gst/datarepo/gstdatareposrc.c

```diff
@@ -373,6 +392,54 @@ gst_data_repo_src_get_file_offset (GstDataRepoSrc * src, guint64 sample_index)
   return offset;
 }
 
+/**
+ * @brief Function to shuffle samples index
+ */
+static void
+gst_data_repo_src_shuffle_samples_index (GstDataRepoSrc * src)
+{
+  guint i, j;
+  gint value_i, value_j;
+  g_return_if_fail (src != NULL);
+  g_return_if_fail (src->shuffled_index_array != NULL);
+
+  GST_LOG_OBJECT (src, "samples index are shuffled");
+
+  /* Fisher-Yates algorithm */
+  /* The last index is the number of samples - 1. */
+  for (i = src->num_samples - 1; i > 0; i--) {
+    j = g_random_int_range (0, src->num_samples);
+    value_i = g_array_index (src->shuffled_index_array, gint, i);
+    value_j = g_array_index (src->shuffled_index_array, gint, j);
+
+    /* shuffled_index_array->data type is gchar * */
+    *(src->shuffled_index_array->data + (sizeof (gint) * i)) = value_j;
+    *(src->shuffled_index_array->data + (sizeof (gint) * j)) = value_i;
+  }
+
+  for (i = 0; i < src->shuffled_index_array->len; i++) {
+    GST_DEBUG_OBJECT (src, "%d -> %d", i,
+        g_array_index (src->shuffled_index_array, gint, i));
```

Comment:
If this loop for debugging, how about run the loop when `silent=FALSE` or `DEBUG=TRUE`?

Reply:
This is output only when the GST_LOG level is 4.
The user needs to export GST_DEBUG=element_name:4 or --gst-debug=element_name:4 in the pipeline to output.